### PR TITLE
Social Links Enhancements

### PR DESCRIPTION
**Pull request by state-hiu**
The social links are no longer hard coded, but added as a list to the settings.py file.  Admins can now easily remove or add their social networks of choice.  This will be useful for managing customization of multiple deployments.

Minor Changes
- prepended a mailto email link.
- context_processors.py defaults SOCIAL_BUTTONS to False if var is not found in settings.py.  Most settings.py files come with SOCIAL_BUTTONS included.
- included https check when sharing url

More importantly, this will open up a simple way to leverage "Web Intents" / APIs to integrate GeoNode services into other systems, such as discovery platforms, url shorteners, etc.  Are there other web intents that should be added to the default settings.py file?

### CODE REVIEW

File: geonode/documents/views.py [hunk 1]
```diff
@@ -83,15 +84,23 @@ def document_detail(request, docid):
...
             Document.objects.filter(id=document.id).update(popular_count=F('popular_count') + 1)
 
+
+        context_dict = {
+            'permissions_json': _perms_info_json(document),
+            'resource': document,
+            'imgtypes': IMGTYPES,
+            'related': related}
+
+        if settings.SOCIAL_ORIGINS:
```
Comment (simod) [inline geonode/documents/views.py]: Would it be possible to avoid the repetition of this piece of code in the three views?

**Comment by garnertb** [comment]
Can you document the `SOCIAL_ORIGINS` here: https://github.com/GeoNode/geonode/blob/master/docs/reference/developers/settings.txt

**Comment by state-hiu** [comment]
SOCIAL_ORIGINS is now documented in https://github.com/GeoNode/geonode/blob/master/docs/reference/developers/settings.txt.

In regards to DRY, the only two keys for context_dict in common among the detail views for layers, maps, and documents are resource and permission_json.

```
'resource': document,
'permissions_json': _perms_info_json(document),
```
I could create a function in utility.py that returns an initial dict with those values.  However, I'd then need to add additional lines of code to append additional key-value pairs as needed, such as imgtypes, related, documents, metadata, is_layer, config, and layers.

**Comment by simod** [comment]
sorry @state-hiu I was referring to this block https://github.com/GeoNode/geonode/pull/1918/files#diff-36bc05580e37aef61b66c480db47cc38R257 until to https://github.com/GeoNode/geonode/pull/1918/files#diff-36bc05580e37aef61b66c480db47cc38R264

**Comment by state-hiu** [comment]
Okay, so should I add that to utils.py or create a new social/utils.py?

**Comment by simod** [comment]
Something in utils should be fine. :) thanks!

**Comment by simod** [comment]
although it's quite social specific....hm dunno if it's worth to create a social/utils.py file just for that but maybe better.

**Comment by state-hiu** [comment]
Just added to utils.py for this PR.  Looking forward, I'm interested in refractoring/modularizing the utility functions during the code sprint.

**Pull request merged**